plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/exporter/QueryInsightsLocalIndexExporter.java

```diff
@@ -0,0 +1,216 @@
+        if (records.size() == 0) {
+            return;
+        }
+        if (checkIfIndexExists()) {
+            bulkRecord(records);
+        } else {
+            // local index not exist
+            initLocalIndex(new ActionListener<>() {
+                @Override
+                public void onResponse(CreateIndexResponse response) {
+                    if (response.isAcknowledged()) {
+                        log.debug(
+                            String.format(Locale.ROOT, "successfully initialized local index %s for query insight.", getIdentifier())
+                        );
+                        try {
+                            bulkRecord(records);
+                        } catch (IOException e) {
+                            log.error(String.format(Locale.ROOT, "fail to ingest query insight data to local index, error: %s", e));
+                        }
+                    } else {
+                        log.error(
+                            String.format(
+                                Locale.ROOT,
+                                "request to created local index %s for query insight not acknowledged.",
+                                getIdentifier()
+                            )
+                        );
+                    }
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    log.error(String.format(Locale.ROOT, "error creating local index for query insight: %s", e));
+                }
+            });
+        }
+    }
+
+    /**
+     * Util function to check if a local OpenSearch Index exists
+     *
+     * @return boolean
+     */
+    private boolean checkIfIndexExists() {
+        ClusterState clusterState = clusterService.state();
+        return clusterState.getRoutingTable().hasIndex(this.getIdentifier());
+    }
+
+    /**
+     * Initialize the local OpenSearch Index for the exporter
+     *
+     * @param listener the listener to be notified upon completion
+     * @throws IOException if an error occurs
+     */
+    private synchronized void initLocalIndex(ActionListener<CreateIndexResponse> listener) throws IOException {
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest(this.getIdentifier()).mapping(getIndexMappings())
+            .settings(Settings.builder().put("index.hidden", false).build());
+        client.admin().indices().create(createIndexRequest, listener);
+    }
+
+    /**
+     * Drop the local OpenSearch Index created by the exporter
+     *
+     * @param listener the listener to be notified upon completion
+     * @throws IOException if an error occurs
+     */
+    private synchronized void dropLocalIndex(ActionListener<AcknowledgedResponse> listener) throws IOException {
+        DeleteIndexRequest deleteIndexRequest = new DeleteIndexRequest(this.getIdentifier());
+        client.admin().indices().delete(deleteIndexRequest, listener);
+    }
```

Comment:
This method looks unused. What is the scenario in which we are planning to use this method?